src/Tribe/Attendee_Registration/Service_Provider.php

```diff
@@ -8,31 +8,148 @@ class Tribe__Tickets__Attendee_Registration__Service_Provider extends tad_DI52_S
-		add_filter( 'tribe_tickets_commerce_paypal_add_to_cart_args', tribe_callback( 'tickets.attendee_registration.meta', 'add_product_delete_to_paypal_url' ), 10, 1 );
+	public function hooks() {
+		// Only run the code if Event Tickets Plus is active.
+		if ( ! class_exists( 'Tribe__Tickets_Plus__Main' ) ) {
+			return;
+		}
+
+		// After priority 45 from register() using this same action.
+		add_action( 'tribe_plugins_loaded', [ $this, 'add_attendee_registration_template_hook' ], 50 );
+
+		add_action( 'init', [ $this, 'add_attendee_registration_shortcode_hook' ] );
+		add_action( 'init', [ $this, 'add_attendee_registration_modal_hook' ] );
+		add_action( 'init', [ $this, 'add_rewrite_tags' ] );
+		add_action( 'tribe_tickets_pre_rewrite', [ $this, 'generate_core_rules' ] );
+
+		add_filter( 'generate_rewrite_rules', [ $this, 'filter_generate' ] );
+		add_filter( 'rewrite_rules_array', [ $this, 'remove_percent_placeholders' ], 25 );
+		add_filter( 'tribe_tickets_commerce_paypal_add_to_cart_args', [ $this, 'add_product_delete_to_paypal_url' ], 10, 1 );
+	}
+
+	/**
+	 * Initialize the template class.
+	 *
+	 * @since TBD
+	 */
+	public function add_attendee_registration_template_hook() {
+		/** @var Tribe__Tickets__Attendee_Registration__Template $make */
+		$make = $this->container->make( 'tickets.attendee_registration.template' );
+
+		$make->hook();
+	}
+
+	/**
+	 * Setup shortcodes for the attendee registration template.
+	 *
+	 * @since TBD
+	 */
+	public function add_attendee_registration_shortcode_hook() {
+		/** @var Tribe__Tickets__Attendee_Registration__Shortcode $make */
+		$make = $this->container->make( 'tickets.attendee_registration.shortcode' );
+
+		$make->hook();
+	}
+
+	/**
+	 * Setup Modal Cart Template.
+	 *
+	 * @since TBD
+	 */
+	public function add_attendee_registration_modal_hook() {
+		/** @var Tribe__Tickets__Attendee_Registration__Modal $make */
+		$make = $this->container->make( 'tickets.attendee_registration.modal' );
+
+		$make->hook();
+	}
+
+	/**
+	 * Add attendee-registration rewrite tag.
+	 *
+	 * @since TBD
+	 */
+	public function add_rewrite_tags() {
+		/** @var Tribe__Tickets__Attendee_Registration__Rewrite $make */
+		$make = $this->container->make( 'tickets.attendee_registration.rewrite' );
+
+		$make->add_rewrite_tags();
+	}
+
+	/**
+	 * Sets up the rules required by Event Tickets.
+	 *
+	 * @since TBD
+	 *
+	 * @param Tribe__Tickets__Attendee_Registration__Rewrite $rewrite The rewrite instance.
+	 */
+	public function generate_core_rules( Tribe__Tickets__Attendee_Registration__Rewrite $rewrite ) {
+		/** @var Tribe__Tickets__Attendee_Registration__Rewrite $make */
+		$make = $this->container->make( 'tickets.attendee_registration.rewrite' );
+
+		$make->generate_core_rules( $rewrite );
+	}
+
+	/**
+	 * Generate the Rewrite Rules.
+	 *
+	 * @since TBD
+	 *
+	 * @param WP_Rewrite $wp_rewrite WordPress rewrite that will be modified.
+	 */
+	public function filter_generate( WP_Rewrite $wp_rewrite ) {
```

Comment:
:no_entry_sign: Please, make sure that a callback to `'generate_rewrite_rules'` filter is always returning some value.

<sub>hash: 444f63411392729343b8974a0697a180</sub>